src/components/shared/MarkerTooltipContents.js

```diff
@@ -121,7 +129,8 @@ function getMarkerDetails(data: MarkerPayload): React.Element<any> | null {
             return (
               <div className="tooltipDetails">
                 {_markerDetail('gcreason', 'Reason', timings.reason)}
-                {timings.nonincremental_reason !== 'None' &&
+                {timings.nonincremental_reason &&
+                  timings.nonincremental_reason !== 'None' &&
```

Comment:
I'm not a big fan of the `condition && result` pattern, although I've not said anything since this was already an established pattern in this file. I would ask to hoist the condition here into a variable above, as this is really hard to reason with and read with the chaining of the `&&` statements.

```js
const hasNonIncrementalReason = timings.nonincremental_reason && timings.nonincremental_reason !== 'None'
```

Reply:
FWIW I also don't like this pattern either. Maybe for a different reason though. I don't line relying on short-circuit behavour of logical operators to do conditions. I'd prefer a ternery operator like in C. This way the programer's intention and the program's semantics are more in-line.